pkg/driver/efs_watch_dog.go

```diff
@@ -39,22 +101,116 @@ type execWatchdog struct {
 	execArg []string
 	// the cmd that is running
 	cmd *exec.Cmd
+	// efs-utils config file path
+	efsUtilsCfgPath string
+	// efs-utils static files path
+	efsUtilsStaticFilesPath string
 	// stopCh indicates if it should be stopped
 	stopCh chan struct{}
 
 	mu sync.Mutex
 }
 
-func newExecWatchdog(cmd string, arg ...string) Watchdog {
+type efsUtilsConfig struct {
+	EfsClientSource string
+}
+
+func newExecWatchdog(efsUtilsCfgPath, efsUtilsStaticFilesPath, cmd string, arg ...string) Watchdog {
 	return &execWatchdog{
-		execCmd: cmd,
-		execArg: arg,
-		stopCh:  make(chan struct{}),
+		efsUtilsCfgPath:         efsUtilsCfgPath,
+		efsUtilsStaticFilesPath: efsUtilsStaticFilesPath,
+		execCmd:                 cmd,
+		execArg:                 arg,
+		stopCh:                  make(chan struct{}),
 	}
 }
 
-func (w *execWatchdog) start() {
+func (w *execWatchdog) start() error {
+	if err := w.setup(GetVersion().EfsClientSource); err != nil {
+		return err
+	}
+
 	go w.runLoop(w.stopCh)
+
+	return nil
+}
+
+func (w *execWatchdog) setup(efsClientSource string) error {
+	if err := w.restoreStaticFiles(); err != nil {
+		return err
+	}
+
+	if err := w.updateConfig(efsClientSource); err != nil {
+		return err
+	}
+	return nil
+}
+
+/**
+	At image build time, static files installed by efs-utils in the config directory, i.e. CAs file, need
+	to be saved in another place so that the other stateful files created at runtime, i.e. private key for
+	client certificate, in the same config directory can be persisted to host with a host path volume.
+	Otherwise creating a host path volume for that directory will clean up everything inside at the first time.
+	Those static files need to be copied back to the config directory when the driver starts up.
+*/
+func (w *execWatchdog) restoreStaticFiles() error {
+	return copyWithoutOverwriting(w.efsUtilsStaticFilesPath, w.efsUtilsCfgPath)
+}
+
+func copyWithoutOverwriting(srcDir, dstDir string) error {
+	if _, err := os.Stat(dstDir); err != nil {
+		return err
+	}
+
+	entries, err := ioutil.ReadDir(srcDir)
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		src := filepath.Join(srcDir, entry.Name())
+		dst := filepath.Join(dstDir, entry.Name())
+
+		if _, err := os.Stat(dst); os.IsNotExist(err) {
+			klog.Infof("Copying %s since it doesn't exist", dst)
+			if err := copyFile(src, dst); err != nil {
+				return err
+			}
+		} else {
+			klog.Infof("Skip copying %s since it exists already", dst)
+		}
+	}
+	return nil
+}
+
+func copyFile(src, dst string) error {
+	source, err := os.Open(src)
```

Comment:
If src is a directory, this code will fail, which is good because we will catch it fast. If for some reason efs-utils in a new version starts persisting more stuff into /etc/amazon/efs.

Reply:
That's a great point. Added a unit test to ensure it failed in this case.